cmd/bucket-metadata-sys.go

```diff
@@ -420,6 +420,31 @@ func (sys *BucketMetadataSys) GetConfig(bucket string) (BucketMetadata, error) {
 	return meta, nil
 }
 
+// IsTierInUse returns true if there exists at least one active ILM rules
+// refering to tierName. Otherwise it returns false.
+//
+// N B this check is potentially racy, i.e a concurrent
+// PutBucketLifecycle/DeleteBucketLifecycle API request could change this
+// shortly after the check. Since, this check is only a best effort attempt to
+// prevent a user from deleting a remote tier that is in active use, this race
+// is acceptable.
+func (sys *BucketMetadataSys) IsTierInUse(tierName string) bool {
+	sys.RLock()
+	defer sys.RUnlock()
+	for _, bucketMeta := range sys.metadataMap {
+		for _, rule := range bucketMeta.lifecycleConfig.Rules {
```

Comment:
you are assuming here that the bucket has a lifecycle config.

Reply:
ah, good catch. I have addressed this by skipping over buckets without lifecycle rules.